### DISCUSSION TITLE
Streamlit deployment

### DISCUSSION THREAD

**Discussion by 11dimnsion** [JoshuaC215/agent-service-toolkit#79] [Q&A] [answered]
This is an awesome project; and I'm using it for rapid prototyping of a bunch of agentic ideas. One this I haven't been able to figure out though is how you managed to get this running in Streamlit cloud. I haven't been able to successfully start the service process from unside of streamlit and I don't see other pipeline options in streamlit. Could you share your approach?

Thanks!

**Comment by JoshuaC215** [comment, marked as answer]
Hey @11dimnsion thank you!

The service is running as an azure web app, not in Streamlit Cloud. I’m pushing the service docker image to docker hub and then deploying it from there.

You can see the automation here, it updates on every push to main now so I don’t have to mess with it:

 https://github.com/JoshuaC215/agent-service-toolkit/blob/main/.github/workflows/deploy.yml

To run yourself you’d need to modify the condition at the top and set the right secrets in your repo. Then once it’s deployed you just need to set the azure URL and the secret header (if desired) in the streamlit app secrets.

It’s been running on the Azure free tier with modest daily traffic from this repo for a couple months and hasn’t cost me anything.

I did set up the app manually in the azure web console initially. I’d like to add terraform or more specific instructions to set it up but haven’t gotten around to it. I imagine it would be fairly straightforward, if you want to document and share back anything you do that would be great!!

Hope this helps, let me know if you have any other questions.

**Reply by JoshuaC215** [reply]
Oh and btw I think the same thing would work on AWS I just happened to pick Azure for this one

**Reply by 11dimnsion** [reply]
Thanks very much! I was able to get everything running pretty smoothly. The biggest hurdle was that I needed to specify '--platform linux/amd64' when building the Dockerfile since was building locally and I am working from a mac with Apple Silicon.

Regarding documentation and sharing, I’ve integrated a working RAG tool and repurposed the left sidebar to include buttons for pre-defined prompts. If either of these features interests you, I’d be happy to put together a clean implementation and submit a pull request.

**Reply by JoshuaC215** [reply]
Nice that’s great to hear!

Adding the sidebar buttons for pre-defined prompts sounds great if you want to put together a PR :)
